src/values/ObjectValue.js

```diff
@@ -280,6 +282,14 @@ export default class ObjectValue extends ConcreteValue {
   // ES2015 classes
   $IsClassPrototype: boolean;
 
+  // We track some internal state as properties on the global object, these should
+  // never be serialized.
+  refuseSerialization: boolean;
+
+  addTemporalAlias(temporalValue: AbstractObjectValue) {
+    this.temporalAlias = temporalValue;
```

Comment:
State invariant that `this.temporalAlias === undefined || this.temporalAlias === temporalValue`.

Reply:
No, the temporalValue should be a differently value at each point where a snapshot is taken.